fix: database path issue when executed from a relative path

When attempting to run with a cwd of outside of the path containing the main.go file:

```bash
    ~/github.com/danstis/go-read-burn    add-base-fetures !1 ······················································································································································· 11:20:17 AM  
❯ go run cmd/go-read-burn/main.go
2022/11/12 11:20:22 Version "0.0.0-DEVELOPMENT"
2022/11/12 11:20:22 open ../../db/secrets.db: no such file or directory
panic: template: pattern matches no files: `views/*.html`
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x188 pc=0x6c636f]

goroutine 1 [running]:
github.com/boltdb/bolt.(*DB).Close(0x0)
	/home/dan/go/pkg/mod/github.com/boltdb/bolt@v1.3.1/db.go:392 +0x4f
panic({0x720940, 0xc000098f50})
	/usr/local/go/src/runtime/panic.go:884 +0x212
text/template.Must(...)
	/usr/local/go/src/text/template/helper.go:26
main.main()
	/home/dan/github.com/danstis/go-read-burn/cmd/go-read-burn/main.go:44 +0x537
exit status 2

```

--- a/.gitignore
+++ b/.gitignore
@@ -26,5 +26,5 @@ _testmain.go
 /.vscode
 
 # Project specific
-/db/*
+*/secrets.db*
 deploy/.env

--- a/cmd/go-read-burn/main.go
+++ b/cmd/go-read-burn/main.go
@@ -14,6 +14,7 @@ import (
 	"github.com/boltdb/bolt"
 	"github.com/danstis/go-read-burn/internal/version"
 	"github.com/gorilla/mux"
+	"github.com/kelseyhightower/envconfig"
 )
 
 var (
@@ -22,12 +23,26 @@ var (
 	templates *template.Template
 )
 
+type Config struct {
+	DBPath string `default:"db/secrets.db"`
+}
+
 // Main entry point for the app.
 func main() {
 	log.Printf("Version %q", version.Version)
 
+	// Read config
+	var config Config
 	var err error
-	db, err = bolt.Open(dbPath, 0644, nil)
+	if err = envconfig.Process("GRB", &config); err != nil {
+		log.Println(err)
+	}
+
+	// Open the DB
+	if err = createDBDir(config.DBPath); err != nil {
+		log.Fatalf("failed to create database directory: %v", err)
+	}
+	db, err = bolt.Open(config.DBPath, 0644, nil)
 	if err != nil {
 		log.Println(err)
 	}
@@ -80,6 +95,11 @@ func main() {
 		log.Println(err)
 	}
 	os.Exit(0)
+}
+
+func createDBDir(p string) error {
+	dir := path.Dir(p)
+	return os.MkdirAll(dir, os.ModePerm)
 }
 
 func IndexHandler(w http.ResponseWriter, r *http.Request) {

--- a/cmd/go-read-burn/main.go
+++ b/cmd/go-read-burn/main.go
@@ -2,6 +2,7 @@ package main
 
 import (
 	"context"
+	"embed"
 	"fmt"
 	"log"
 	"net/http"
@@ -17,8 +18,13 @@ import (
 	"github.com/kelseyhightower/envconfig"
 )
 
+//go:embed all:views/*
+var views embed.FS
+
+//go:embed static/*
+var static embed.FS
+
 var (
-	dbPath    = path.Join("..", "..", "db", "secrets.db")
 	db        *bolt.DB
 	templates *template.Template
 )
@@ -52,11 +58,11 @@ func main() {
 	r.HandleFunc("/", IndexHandler)
 	r.HandleFunc("/create", CreateHandler).Methods("POST")
 	r.HandleFunc("/get/{key}", SecretHandler)
-	s := http.StripPrefix("/static/", http.FileServer(http.Dir("./static/")))
+	s := http.StripPrefix("/static/", http.FileServer(http.FS(static)))
 	r.PathPrefix("/static/").Handler(s)
 	http.Handle("/", r)
 
-	templates = template.Must(template.ParseGlob("views/*.html"))
+	templates = template.Must(template.ParseFS(views, "views/*.html"))
 
 	srv := &http.Server{
 		Handler:      r,

--- a/go.mod
+++ b/go.mod
@@ -5,6 +5,7 @@ go 1.19
 require (
 	github.com/boltdb/bolt v1.3.1
 	github.com/gorilla/mux v1.8.0
+	github.com/kelseyhightower/envconfig v1.4.0
 	github.com/uptrace/opentelemetry-go-extra/otelzap v0.1.17
 	go.opentelemetry.io/otel v1.10.0
 	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.10.0

